platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigDataFactory.java

```diff
@@ -29,10 +33,12 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * Internal factory for config data objects. See {@link Configuration#getConfigData(Class)} for a detailed description on
```

Comment:
Really needs a pointer to the design doc that includes this reflection-heavy class.

Reply:
What do you mean?

Reply:
well, the way this factory is using reflection on stuff to get all kinds of information to populate the config - it's interesting and non-obvious and not immediately clear why this approach is taken.  So if there was a design doc that describes this approach, leave a link, otherwise - maybe just a sentence or two here would describe what's going on?  the code is not very long but it is compact&intricate

Reply:
ok, got it

Reply:
issue created https://github.com/hashgraph/hedera-services/issues/6228